packages/react-core/src/components/Label/Label.tsx

```diff
@@ -254,7 +257,13 @@ export const Label: React.FunctionComponent<LabelProps> = ({
   };
 
   let labelComponentChild = (
-    <LabelComponentChildElement {...labelComponentChildProps}>{content}</LabelComponentChildElement>
+    <LabelComponentChildElement
+      type={LabelComponentChildElement === 'button' ? 'button' : undefined}
+      onClick={onLabelClick}
```

Comment:
It could make sense to only pass on onLabelClick if the `LabelComponentChildElement === 'button'`, as we probably don't want to allow a label to both trigger an action on click *and* redirect a user to another page. That would also mean updating prop descriptions to mention not using `href` if `onLabelClick` is passed and vice versa

Reply:
+1

Also I think having this logic as part of the `labelComponentChildProps` definition would be a bit cleaner of an approach.